workflows/multiomics/integration/main.nf

```diff
@@ -58,13 +57,34 @@ workflow run_wf {
         "obsm_output": "obsm_umap"
       ]
     )
-
+    | map { tup ->
+      assert tup.size() >= 3: "Event should have length 3 or greater."
+      id = tup[0]
+      current_args = tup[1]
+      other_args = tup[2]
+      passthrough = tup.drop(3)
+      // If obs_covariates is not set or emtpy, harmony will not be run
+      // In this case, the layer that find_neighbour uses should not originate from harmonypy but from pca
+      if (!other_args.integration.obs_covariates || 
+        (other_args.integration.obs_covariates && other_args.integration.obs_covariates.empty)) {
+        other_args.neighbors.put("obsm_input", other_args.pca.obsm_output)
+      } else {
+        other_args.neighbors.put("obsm_input", other_args.integration.obsm_output)
+      }
```

Comment:
```suggestion
      def obs_cov = other_args.integration.obs_covariates ?: []
      if (obs_cov.empty) {
        other_args.neighbors.put("obsm_input", other_args.pca.obsm_output)
      } else {
        other_args.neighbors.put("obsm_input", other_args.integration.obsm_output)
      }
```

Reply:
Be careful, this changes the original `tuple` due to pass by reference instead of pass by value.

Reply:
See other suggestion